### ISSUE TITLE
std/emulated/field_assert.go  IsZero() does not work as expected  

### ISSUE THREAD

**Issue by readygo67** [Consensys/gnark#980]
std/emulated/field_assert.go  IsZero() does not work as expected  

## Description
std/emulated/field_assert.go  IsZero() does not work as expected, original code  is below,  res is supposed impacted by all  limbs, but unfortunately, the for loop does not impact res.

```
func (f *Field[T]) IsZero(a *Element[T]) frontend.Variable {
	ca := f.Reduce(a)
	f.AssertIsInRange(ca)
	res := f.api.IsZero(ca.Limbs[0])
	for i := 1; i < len(ca.Limbs); i++ {
		f.api.Mul(res, f.api.IsZero(ca.Limbs[i]))
	}
	return res
}

```

## Expected Behavior and Actual Behavior
For the below TestZeroCircuit_NonZero,  It should be NonZero, But IsZero() return true
```
type ZeroCircuit[T FieldParams] struct {
	Var    Element[T]
	IsZero frontend.Variable `gnark:",public"`
}

func (c *ZeroCircuit[T]) Define(api frontend.API) error {
	f, err := NewField[T](api)
	if err != nil {
		return err
	}
	res := f.IsZero(&c.Var)
	f.api.AssertIsEqual(res, c.IsZero)
	return nil
}

func TestZeroCircuit_NonZero(t *testing.T) {
	assert := test.NewAssert(t)
	type T = BLS12381Fp
	var circuit, witness ZeroCircuit[T]
	el := Element[T]{
		Limbs:    []frontend.Variable{0, 1, 0, 1, 0, 1},
		overflow: 0,
		internal: true,
	}
	witness.Var = el
	witness.IsZero = 0

	err := test.IsSolved(&circuit, &witness, testCurve.ScalarField())
	assert.NoError(err)

	_, err = frontend.Compile(testCurve.ScalarField(), r1cs.NewBuilder, &circuit)
	assert.NoError(err)
}
```

## Possible Fix
```
func (f *Field[T]) IsZero(a *Element[T]) frontend.Variable {
	ca := f.Reduce(a)
	f.AssertIsInRange(ca)
	res := f.api.IsZero(ca.Limbs[0])
	for i := 1; i < len(ca.Limbs); i++ {
		res = f.api.Mul(res, f.api.IsZero(ca.Limbs[i]))
	}
	return res
}
```

**Comment by ivokub** [comment]
Indeed, you are right. We would be excited to accept your contribution as a PR! Otherwise, I can implement the fix, but this would leave you out of the list of [code contributors](https://github.com/Consensys/gnark/graphs/contributors)

**Comment by readygo67** [comment]
Great! I will create a PR to fix this problem soon.

**Comment by ivokub** [comment]
> Great! I will create a PR to fix this problem soon.

Perfect! Will be looking forward to it!

**Comment by readygo67** [comment]
@ivokub  It looks like I have no right to push my fix, could you do me a favor to open write permission for me? thanks

**Comment by ivokub** [comment]
> @ivokub It looks like I have no right to push my fix, could you do me a favor to open write permission for me? thanks

I see you have solved it by forking and pushing a PR from there. That was actually what I would have proposed - we do not grant direct write access but rather pull the contributions from forks.

**Comment by ivokub** [comment]
Fixed through #1002 1002
